SYCL/ESIMD/api/functional/ctors/ctor_load_core.cpp

```diff
@@ -0,0 +1,233 @@
+    if constexpr (NumElems == 1) {
+      for (size_t i = 0; i < ref_data.size(); ++i) {
+        passed = run_verification(queue, {ref_data[i]}, data_type);
+      }
+    } else {
+      passed = run_verification(queue, ref_data, data_type);
+    }
+    return passed;
+  }
+
+private:
+  bool run_verification(sycl::queue &queue, const std::vector<DataT> &ref_data,
+                        const std::string &data_type) {
+    assert(ref_data.size() == NumElems &&
+           "Reference data size is not equal to the simd vector length.");
+
+    bool passed = true;
+
+    const size_t alignment_value =
+        AlignmentT::template get_size<DataT, NumElems>();
+    const size_t container_extra_size = alignment_value / sizeof(DataT) + 1;
+    const size_t offset = 1;
+
+    shared_allocator<DataT> allocator(queue);
+    shared_vector<DataT> result(NumElems, allocator);
+    shared_vector<DataT> shared_ref_data(NumElems + container_extra_size +
+                                             offset,
+                                         shared_allocator<DataT>(queue));
+
+    const size_t object_size = NumElems * sizeof(DataT);
+    size_t buffer_size = object_size + container_extra_size * sizeof(DataT);
+
+    // When we allocate USM there is a high probability that this memory will
+    // have stronger alignment that required. We increment our pointer by fixed
+    // offset value to avoid bigger alignment of USM shared.
+    // The std::align can provide expected alignment on the small values of an
+    // alignment.
+    void *ref = shared_ref_data.data() + offset;
+    if (std::align(alignment_value, object_size, ref, buffer_size) == nullptr) {
+      return false;
+    }
+    DataT *const ref_aligned = static_cast<DataT *>(ref);
+
+    for (size_t i = 0; i < NumElems; ++i) {
+      ref_aligned[i] = ref_data[i];
+    }
+
+    queue.submit([&](sycl::handler &cgh) {
+      DataT *const out = result.data();
+
+      cgh.single_task<
+          Kernel_for_load_ctor<DataT, NumElems, TestCaseT, AlignmentT>>(
+          [=]() SYCL_ESIMD_KERNEL {
+            const auto alignment = AlignmentT::get_value();
+            TestCaseT::template call_simd_ctor<DataT, NumElems>(ref_aligned,
+                                                                out, alignment);
+          });
+    });
+    queue.wait_and_throw();
+
+    for (size_t i = 0; i < result.size(); ++i) {
+      if (!are_bitwise_equal(ref_data[i], result[i])) {
+        passed = false;
+
+        const auto description =
+            ctors::TestDescription<DataT, NumElems, TestCaseT>(
+                i, result[i], ref_data[i], data_type);
+        log::fail(description);
+      }
+    }
+
+    return passed;
+  }
+};
+
+int main(int, char **) {
+  sycl::queue queue(esimd_test::ESIMDSelector{},
+                    esimd_test::createExceptionHandler());
+
+  bool passed = true;
+  const auto types = get_tested_types<tested_types::all>();
+  const auto dims = get_all_dimensions();
```

Comment:
Nit for future: please rename `dimensions` to `vector_lengths` or `sizes`

Reply:
`dimensions` will be changed to one of suggested names for all tests in the one of next PR`